libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -1043,6 +1043,9 @@ bool RC_Channel::run_aux_function(aux_func_t ch_option, AuxSwitchPos pos, AuxFun
 
 bool RC_Channel::do_aux_function(const aux_func_t ch_option, const AuxSwitchPos ch_flag)
 {
+#if AP_SCRIPTING_ENABLED
+    rc().set_aux_cached(ch_option, ch_flag);
```

Comment:
I don't think this works in every case because of the vehicle overrides. If the function is one the vehicle understands we will never hit this?

Reply:
I think this needs to be in `run_aux_function`

Reply:
thanks, fixed!